app/api/api/announcements_api.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module API
+  class AnnouncementsAPI < API::Base
+    get '/' do
+      std_result Announcement.all.order(id: :desc), filter: 'HNLMH'
+    end
+
+    get 'active' do
+      std_result Announcement.where('expiry > ?', DateTime.now).order(id: :desc), filter: 'HNLMH'
+    end
+
+    get 'expired' do
+      std_result Announcement.where('expiry < ?', DateTime.now).order(id: :desc), filter: 'HNLMH'
+    end
+
+    before do
```

Comment:
Maybe makes sense to put the `before` block at the top of the file

Reply:
Deliberate - the `before` block only applies to the subsequent route (in this case, the `post 'create'` route). We don't want to `authenticate_user!` for every route in the file, only for write routes.